lib/puppet/type.rb

```diff
@@ -1316,6 +1318,7 @@ def pathbuilder
       and the second run will log the edit made by Puppet.)"
 
     validate do |list|
+      Puppet.deprecation_warning(_("The `audit` metaparameter is deprecated and will be ignored in a future release."))
```

Comment:
I think this will cause one deprecation warning to be issue for each instance of `audit` in a resource. I'm wondering if this should be a docs only deprecation or a "warn once" type of deprecation. /cc @nfagerlund @ahpook